python/tvm/autotvm/config_library.py

```diff
@@ -0,0 +1,279 @@
+            The config for the specified task. Returns None if no config was
+            found.
+
+        """
+        config_file = self.get_config_file(target, create=False)
+        if config_file:
+            with open(config_file) as f:
+                configs = json.load(f)
+                workload_key = str(workload)
+                if workload_key in configs:
+                    return configs[workload_key]
+
+        return None
+
+    def get_config_file(self, target, create=True):
+        """Return the config file path associated with a given target"""
+        with open(self.library_index, "r+") as f:
+            config_index = json.load(f)
+
+        target_file_name = self._get_target_file_name(target)
+        root = config_index["root"]
+        config_files = config_index["targets"]
+        if target_file_name in config_files:
+            return config_files[target_file_name]
+        elif create:  # Create the file if it's not already in the index
+            with open(self.library_index, "w") as f:
+                config_file_name = target_file_name + ".configs"
+                config_file = os.path.join(root, config_file_name)
+                with open(config_file, "w") as g:
+                    json.dump({}, g)
+
+                config_index["targets"][target_file_name] = config_file
+                json.dump(config_index, f, indent=4)
+                return config_file
+
+        return None
+
+    @staticmethod
+    def _get_target_file_name(target):
+        """Create a file name from a TVM target string."""
+        options = target.split(" ")
+        sorted_options = [options[0]] + sorted(options[1:])
+        return "-".join(sorted_options)
+
+
+class NoIndent(object):
+    def __init__(self, value):
+        self.value = value
+
+
+# TODO @mbarrett97 Find a more efficient way to pretty print the JSON
+class NoIndentEncoder(json.JSONEncoder):
+    """JSON pretty printing class to print configs on one line."""
+    marker = "~@{}@~"
+
+    def default(self, obj):
+        if isinstance(obj, NoIndent):
+            return self.marker.format(id(obj))
+        else:
+            super(NoIndentEncoder, self).default(obj)
+
+    def encode(self, obj):
+        json_obj = super(NoIndentEncoder, self).encode(obj)
+        offset = 0
+        while offset != -1:
+            no_indent_index_start = json_obj.find("~@", offset)
+            no_indent_index_stop = json_obj.find("@~", offset)
+            if no_indent_index_start == -1:
+                break
+
+            no_indent_id = int(json_obj[no_indent_index_start+2:no_indent_index_stop])
+            no_indent_obj = PyObj_FromPtr(no_indent_id)
+            no_indent_json = json.dumps(no_indent_obj.value)
+            json_obj = json_obj.replace(
+                '"{}"'.format(self.marker.format(no_indent_id)),
+                no_indent_json,
+            )
+            offset = no_indent_index_stop + 2
+
+        return json_obj
```

Comment:
If the purpose of implementing this class is just to put multiple JSON objects to the same file, I would suggest using one nested JSON object. For example:

Current:
out.json
{ 1st config }
{ 2nd config }

Proposed:
out.json
[{1st config},
 {2nd config}]

Reply:
I store the configs currently as 
{ 
  [workload_str]: config,
  [workload_str]: config,
}
to make it simpler to query by workload. The issue I was finding was when using the JSON indent option, the configs got massively expanded as they are themselves heavily nested objects. This encoder makes it so the configs ignore the indent value and are printed on one line. I'm very open to an alternative method here as it seems quite messy!

Reply:
Why we need to indent the JSON file if we didn't expect people to read it?

Reply:
If no one cares about human readability I'm happy to drop this. I was mostly trying to preserve the appearance of the existing format where it's 1 config per line.